src/mvs_eland/C0_data_processing.py

```diff
@@ -1699,35 +1699,49 @@ def add_maximum_cap(dict_values, group, asset, subasset=None):
     asset: str
         asset name
     subasset: str
-        subasset name. Default: None.
+        subasset name.
+        Default: None.
 
     Returns
     -------
 
     """
     if subasset is None:
-        dict = dict_values[group][asset]
+        asset_dict = dict_values[group][asset]
     else:
-        dict = dict_values[group][asset][subasset]
-    if MAXIMUM_CAP in dict:
-        # check if maximumCap is greater that installedCap
-        if dict[MAXIMUM_CAP][VALUE] is not None:
-            if dict[MAXIMUM_CAP][VALUE] < dict[INSTALLED_CAP][VALUE]:
+        asset_dict = dict_values[group][asset][subasset]
+    if MAXIMUM_CAP in asset_dict:
+        # Check if a maximumCap is defined
+        if asset_dict[MAXIMUM_CAP][VALUE] is not None:
+            # adapt maximumCap of non-dispatchable sources
+            if (
+                group == ENERGY_PRODUCTION
+                and asset_dict[FILENAME] != None
```

Comment:
@SabineHaas , @smartie2076 - now I encounter an error when using maximum cap from the server because then `FILENAME` is not part of the asset_dict. Is there other ways to flag assets as dispatchable besides the presence of FILENAME value? The existence of `TIMESERIES` in dict_asset would be more accurate no?

Reply:
I made a change in #939

Reply:
>  Is there other ways to flag assets as dispatchable besides the presence of FILENAME value? The existence of TIMESERIES in dict_asset would be more accurate no?

I am sorry, I do not know this from the top of my head anymore.